resources/generate_spec_files.py

```diff
@@ -1,101 +1,182 @@
+                            break
+                # Process the schema file
+                process_schema_file(schema_file, header_file, var_names, var_contents, chunk_size)
+        generate_header_file(ns, header_file, var_names, var_contents)
 
-                    # Concatenate chunks at compile-time
-                    with open(header_file, 'a') as fo:
-                        fo.write(f'constexpr std::array<std::string_view, {len(chunks)}> {var_name}_parts = {{{", ".join(chunk_var_names)}}};\n')
-                        fo.write(f'constexpr std::size_t {var_name}_total_length = []() {{\n')
-                        fo.write(f'    std::size_t length = 0;\n')
-                        fo.write(f'    for (const auto& part : {var_name}_parts) {{\n')
-                        fo.write(f'        length += part.size();\n')
-                        fo.write(f'    }}\n')
-                        fo.write(f'    return length;\n')
-                        fo.write(f'}}();\n')
-                        fo.write(f'constexpr auto {var_name}_concatenate = []() {{\n')
-                        fo.write(f'    std::array<char, {var_name}_total_length + 1> result{{}};\n')
-                        fo.write(f'    std::size_t pos = 0;\n')
-                        fo.write(f'    for (const auto& part : {var_name}_parts) {{\n')
-                        fo.write(f'        for (char c : part) {{\n')
-                        fo.write(f'            result[pos++] = c;\n')
-                        fo.write(f'        }}\n')
-                        fo.write(f'    }}\n')
-                        fo.write(f'    result[pos] = \'\\0\';\n')
-                        fo.write(f'    return result;\n')
-                        fo.write(f'}}();\n')
-                        fo.write(f'constexpr std::string_view {var_name}({var_name}_concatenate.data(), {var_name}_concatenate.size() - 1);\n\n')
-                    var_names.append(var_name)
-                else:
-                    with open(header_file, 'a') as fo:
-                        var_name = s['source'].replace('.yaml', '').replace('.', '_')
-                        fo.write(f'constexpr std::string_view {var_name} = R"delimiter(\n{json.dumps(spec, separators=(',', ':'))})delimiter";\n\n')
-                        var_names.append(var_name)
-                    
-                    
+def process_schema_files(schema_dir: Path, output_dir: Path, chunk_size: int) -> None:
+    """
+    Process schema files in the specified directory and generate corresponding C++ header files.
 
+    Args:
+        schema_dir (Path): Directory containing the schema files.
+        output_dir (Path): Directory to output the generated header files.
+        chunk_size (int): Size of the chunks for splitting large JSON strings.
+    """
+    logger.info(f"Starting to process schema files in directory: {schema_dir}")
+    for file in schema_dir.rglob(r"*namespace.*"):
+        if file.suffix in ['.yaml', '.yml', '.json']:
```

Comment:
```suggestion
        if file.suffix in ['.yaml', '.yml']:
```

See comment above about supporting the json files. I was running into errors when pointing to the .json directory anyway.